components/helm-broker/cmd/preupgrade/main.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"flag"
+	"strings"
+
+	"github.com/kyma-project/kyma/components/helm-broker/platform/logger"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// pre-upgrade kyma release 0.7 -> 0.8
+func main() {
+	verbose := flag.Bool("verbose", false, "specify if log verbosely loading configuration")
+	flag.Parse()
+	cfg, err := Load(*verbose)
+	fatalOnError(err)
+
+	// creates the in-cluster k8sConfig
+	k8sConfig, err := newRestClientConfig(cfg.KubeconfigPath)
+	fatalOnError(err)
+
+	// creates the clientset
+	clientset, err := kubernetes.NewForConfig(k8sConfig)
+	fatalOnError(err)
+
+	log := logger.New(&cfg.Logger)
+
+	log.Info("Searching for additional repositories")
+	deployments, err := clientset.AppsV1().Deployments(cfg.Namespace).List(v1.ListOptions{
+		LabelSelector: "app=helm-broker",
+	})
+	fatalOnError(err)
+	if len(deployments.Items) > 1 {
+		log.Warn("Found more then one helm-broker deployments")
+	}
+
+	reposURLs := ""
+	for _, deploy := range deployments.Items {
+		for _, container := range deploy.Spec.Template.Spec.Containers {
+			for _, env := range container.Env {
+				if env.Name == "APP_REPOSITORY_URLS" {
+					reposURLs = env.Value
+				}
+			}
+		}
+	}
+	newURLs := strings.Join(strings.Split(reposURLs, ";"), "\n")
```

Comment:
filter URLs - default repo url must be removed from the list

Reply:
fixed